subtests/docker_cli/systemd_in_container/test.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+#
+# Test systemd in containers
+#
+
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 IMAGE-NAME"
+    exit 1
+fi
+image="$1"
+
+# Exit status. Default 0 (success), but set to 1 on any subtest failure
+rc=0
+
+# Main code. Run a command whose output is expected to be a single line,
+# and the first field is a security context (user:role:type:level); verify
+# that command generates exactly one line of output; extract the <type>
+# and <level> fields; and check against a list of one or more expected
+# values. Assume that there's no overlap between valid types & ranges.
+check_output() {
+    local name="$1"; shift
+    local command="$1"; shift
+
+    $command < /dev/ptmx > /tmp/systemd$$.log 2>&1 &
+    sleep 30
```

Comment:
This adds 30 seconds to every autotest run, even if the container starts in two seconds. See below for a version that loops and exits more promptly. It's also inline.

```bash
...
image="$1"

# Start a container running systemd. Clean it up when we exit.
cid=$(docker run --detach -ti $image /usr/sbin/init)
if [ $? -ne 0 ]; then
    echo "FAIL: could not start docker container"
    exit 1
fi
trap 'status=$?; docker rm -f $cid >/dev/null; exit $status' 0

# Exit as soon as we see the following indication of success
expect="Started Journal Service"
wait_until=$(expr $SECONDS + 30)

while [ $SECONDS -le $wait_until ]; do
    if docker logs $cid | grep -q "$expect"; then
        echo "PASS"
        exit 0
    fi

    # Abort if container is no longer running
    status=$(docker inspect --format '{{.State.Running}}' $cid)
    if [ "$status" != "true" -o $? -ne 0 ]; then
        echo "FAIL: docker container exited unexpectedly"
        echo "docker logs:"
        docker logs $cid
        exit 1
    fi

    sleep 1
done

echo "FAIL: did not find expected string '$expect' in docker logs:"
docker logs $cid

exit 1
```